src/bin/coldsnap/main.rs

```diff
@@ -266,6 +268,27 @@ struct DownloadArgs {
     no_progress: bool,
 }
 
+
+/// Turn a user-specified tag string into a Tag object. Tags must start with 'KEY=' and
+/// denote the tag value with ',Value='.
+fn tag_from_str(input: &str) -> std::result::Result<Tag, String> {
+    const KEY_DELIMITER: &str = "Key=";
+    const VALUE_DELIMITER: &str = ",Value=";
+
+    if !input.starts_with(KEY_DELIMITER) {
+        return Err(format!("Tag inputs must start with '{KEY_DELIMITER}'").to_string());
+    }
+
+    if !input.contains(VALUE_DELIMITER) {
+        return Err(format!("Tag inputs must contain value entry with '{VALUE_DELIMITER}'").to_string());
+    }
+
+    //We have already validated the input contains both delimeters so there is no panic risk for unwrapping directly.
+    let k_v: (&str, &str) = input.split_once(KEY_DELIMITER).unwrap().1.split_once(VALUE_DELIMITER).unwrap();
+
+    Ok(Tag::builder().key(k_v.0).value(k_v.1).build())
+}
+
```

Comment:
Some quick tests here would be good, if only to clarify what's currently accepted.
```
#[cfg(test)]
mod test {
    use super::*;

    #[test]
    fn good_tag_inputs() {
        for input in ["Key=A,Value=B", "Key=A,Value="] {
            assert!(tag_from_str(input).is_ok());
        }
    }

    #[test]
    fn bad_tag_inputs() {
        for input in [
            "Key=A",
            ",Value=B",
            "Key=,Value=B",
            "Kay=A,Key=A,Value=B",
        ] {
            assert!(tag_from_str(input).is_err());
        }
    }
}
```